core/dbt/parser/schemas.py

```diff
@@ -351,6 +352,23 @@ def create_test_node(
         column_name: Optional[str],
     ) -> ParsedSchemaTestNode:
 
+        HASH_LENGTH = 10
+
+        def get_hashable_md(
+            data: Union[str, int, float, List, Dict]
+        ) -> Union[str, List, Dict]:
+            if type(data) == dict:
+                return {k: get_hashable_md(data[k]) for k in sorted(data.keys())}  # type: ignore
+            elif type(data) == list:
+                return [get_hashable_md(val) for val in data]  # type: ignore
+            else:
+                return str(data)
+
+        hashable_metadata = repr(get_hashable_md(test_metadata))
+        hash_list = filter(None, [name, hashable_metadata])
```

Comment:
I don't think I understand this line. Using None for our filter function means it's using the identity function, so either these values are booleans, or no filtering is occurring here.

Reply:
Ah, good catch.  It's a hold-over from a previous iteration where we picked specific items from test_metadata to hash instead of hashing the whole thing.  It was basically my way of DRYing up all the "dummy_value"s in a list of this:
```
hash_list.append(test_metadata.get("thing_1", "dummy_value"))
hash_list.append(test_metadata.get("thing_2", "dummy_value"))
hash_list.append(test_metadata.get("thing_3", "dummy_value"))
hash_list.append(test_metadata.get("thing_4", "dummy_value"))
```

I don't _think_ `name` or `hashable_metadata` could ever be `None` (test running right now to confirm)

Reply:
Confirmed and updated.